crates/uv-cli/src/lib.rs

```diff
@@ -2753,6 +2770,83 @@ pub struct TreeArgs {
     pub python: Option<String>,
 }
 
+#[derive(Args)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct ExportArgs {
+    /// The format to which `uv.lock` should be exported.
+    ///
+    /// At present, only `requirements.txt` is supported.
+    #[arg(long, value_enum, default_value_t = ExportFormat::default())]
+    pub format: ExportFormat,
+
+    /// Export the dependencies for a specific package in the workspace.
+    ///
+    /// If not in a workspace, or if the workspace member does not exist, uv
+    /// will exit with an error.
+    #[arg(long)]
+    pub package: Option<PackageName>,
+
+    /// Include optional dependencies from the extra group name.
+    ///
+    /// May be provided more than once.
+    #[arg(long, conflicts_with = "all_extras", value_parser = extra_name_with_clap_error)]
+    pub extra: Option<Vec<ExtraName>>,
+
+    /// Include all optional dependencies.
+    #[arg(long, conflicts_with = "extra")]
+    pub all_extras: bool,
+
+    #[arg(long, overrides_with("all_extras"), hide = true)]
+    pub no_all_extras: bool,
+
+    /// Include development dependencies.
+    #[arg(long, overrides_with("no_dev"), hide = true)]
+    pub dev: bool,
+
+    /// Omit development dependencies.
+    #[arg(long, overrides_with("dev"))]
+    pub no_dev: bool,
+
+    /// Assert that the `uv.lock` will remain unchanged.
+    ///
+    /// Requires that the lockfile is up-to-date. If the lockfile is missing or
+    /// needs to be updated, uv will exit with an error.
+    #[arg(long, conflicts_with = "frozen")]
+    pub locked: bool,
+
+    /// Assert that a `uv.lock` exists, without updating it.
```

Comment:
Can we write a bit more here? What are the implications for a user running an export here?  e.g.

> Do not update the `uv.lock` before exporting.
>
> If a `uv.lock` does not exist, uv will exit with an error.

Did you copy these from somewhere? I thought I touched most of these up to say stuff like that.

Reply:
The `LockArgs` one.